src/install/install.zig

```diff
@@ -2000,23 +2001,23 @@ pub const PackageManager = struct {
         return this.allocator.create(NetworkTask) catch @panic("Memory allocation failure creating NetworkTask!");
     }
 
-    fn allocGitHubURL(this: *const PackageManager, repository: *const Repository) !string {
+    fn buildGitHubURL(this: *const PackageManager, buf: []u8, repository: *const Repository) string {
         var github_api_domain: string = "api.github.com";
         if (this.env.map.get("GITHUB_API_DOMAIN")) |api_domain| {
             if (api_domain.len > 0) {
                 github_api_domain = api_domain;
             }
         }
-        return try std.fmt.allocPrint(
-            this.allocator,
+        return std.fmt.bufPrintZ(
+            buf,
             "https://{s}/repos/{s}/{s}/tarball/{s}",
             .{
                 github_api_domain,
                 this.lockfile.str(&repository.owner),
                 this.lockfile.str(&repository.repo),
                 this.lockfile.str(&repository.committish),
             },
-        );
+        ) catch unreachable;
```

Comment:
This isn't actually unreachable

It would be a buffer overflow

Reply:
if combo of tag + repo + organization exceeded `1024` bytes (macOS) or `4096` bytes (Linux), then this buffer would overflow

Reply:
fixed in https://github.com/oven-sh/bun/pull/2520/commits/3736c55d43ef4e9340160b57f652a89bebb43ca9